app/services/cfe/submit_proceedings_service.rb

```diff
@@ -0,0 +1,24 @@
+module Cfe
+  class SubmitProceedingsService < BaseService
+    def call(cfe_assessment_id)
+      applicant_form = ApplicantForm.from_session(@session_data)
+      proceeding_type = if estimate.use_legacy_proceeding_type?
+                          applicant_form.legacy_proceeding_type
+                        elsif relevant_form?(:matter_type)
+                          matter_type_form = MatterTypeForm.from_session(@session_data)
+                          matter_type_form.proceeding_type
+                        else
+                          ApplicantForm::PROCEEDING_TYPES[:other]
+                        end
+
+      proceeding_types = [
```

Comment:
did you move this logic here to make the testing easier?

Reply:
Yes, and to make CfeConnection more consistent: in every other case, the connection blindly turns what it's given into the payload of a request. Here it used to have knowledge of what should be _in_ that request, for no good reason.